### PR TITLE
Fix deprecation warnings with Gdk w/ Screen and monitor

### PR DESCRIPTION
**Pull request by ItzSwirlz**
Using gdk_screen and gdk_display_get_default fixes.

### CODE REVIEW

File: libcscreensaver/cs-screen-x11.c [hunk 1]
```diff
@@ -393,7 +393,7 @@ reload_monitor_infos (CsScreen *screen)
 
     apply_scale_factor (screen->monitor_infos,
                         screen->n_monitor_infos,
-                        gdk_screen_get_monitor_scale_factor (screen->gdk_screen, PRIMARY_MONITOR));
+                        gdk_monitor_get_scale_factor (PRIMARY_MONITOR));
```
Comment (mtwebster) [inline libcscreensaver/cs-screen-x11.c:396]: You need to use `gdk_display_get_monitor (gdk_display, PRIMARY_MONITOR)` and call this on the returned GdkMonitor

File: libcscreensaver/cs-screen-x11.c [hunk 2]
```diff
@@ -689,9 +689,9 @@ cs_screen_get_mouse_monitor (CsScreen *screen)
                    &win_x_return,
                    &win_y_return,
                    &mask_return);
-    gdk_error_trap_pop_ignored ();
+    gdk_x11_display_error_trap_pop_ignored (gdk_display);
 
-    scale_factor = gdk_screen_get_monitor_scale_factor (screen->gdk_screen, 0);
+    scale_factor = gdk_monitor_get_scale_factor (screen->gdk_screen);
```
Comment (mtwebster) [inline libcscreensaver/cs-screen-x11.c:694]: same thing here

**Comment by ItzSwirlz** [comment]
If using ```gdk_display_get_monitor``` works for you, then I'll switch it.

**Comment by mtwebster** [comment]
No, I mean you need a GdkMonitor object to use gdk_monitor_get_scale_factor:
```
gdk_monitor_get_scale_factor (gdk_display_get_monitor (gdk_display, PRIMARY_MONITOR));
```
The same for the other call below (line 694).

Also, gdk_display_get_default is a function:
https://github.com/linuxmint/cinnamon-screensaver/pull/351/files#diff-b7f7d2594103c0f36dca03a45a046b08fb865e804345a193352e36e20556c673R111
should be:
```
gdk_x11_display_error_trap_push (gdk_display_get_default ());
```

These problem spots should all give warnings during the build (this is why I tend to want to hide deprecation warnings in other projects, so I can catch the 'real' problems).

**Comment by ItzSwirlz** [comment]
```
These problem spots should all give warnings during the build (this is why I tend to want to hide deprecation warnings in other projects, so I can catch the 'real' problems).
```

In that case, do you want to come back to this later? (This allows me to study GDK and that stuff to ensure that y'know, I don't mess up)

**Comment by mtwebster** [comment]
We can revisit this if the issues are corrected. Closing for now.

**Pull request closed without merging**